web-wallet/src/style/dusk-components/button.css

```diff
@@ -38,130 +90,17 @@
   text-overflow: ellipsis;
 }
 
-.dusk-button:enabled:active,
-.dusk-anchor-button:not(.dusk-anchor-button--disabled):active {
-  transform: scale(0.98, 0.98);
```

Comment:
Waiting on @laremas to confirm, my understanding is that hover and active states are the same.

Reply:
mmm, not sure i get your question. [Here](https://www.figma.com/file/mkLSb0BV2jJa5blFj8VxUJ/Dusk-2.1-(variables)?type=design&node-id=192%3A1120&mode=design&t=QQ1Ga8zLizzdqmYW-1) are the buttons states. Hover state is different from Active.
<img width="512" alt="Screenshot 2024-04-09 at 12 16 29" src="https://github.com/dusk-network/rusk/assets/1636833/e3a3470a-85e7-4682-95e6-29b53285cd89">

Reply:
> mmm, not sure i get your question. [Here](https://www.figma.com/file/mkLSb0BV2jJa5blFj8VxUJ/Dusk-2.1-(variables)?type=design&node-id=192%3A1120&mode=design&t=QQ1Ga8zLizzdqmYW-1) are the buttons states. Hover state is different from Active. 

Maybe you're referring to the "focused" state?

By "active" we mean "pressed": think of a toggle button where the button stays pressed.

@nortonandreev we need the active state for that one too in the styles.

Also, as we're correcting our BEM classes, this

```js
type === "toggle" && active ? "dusk-button--type--toggle--active" : ""
```

should probably become:

```js
type === "toggle" && active ? "dusk-button--active" : ""
```

Reply:
> Maybe you're referring to the "focused" state?

It looks like it, also the border to the state was just added – I don't think it was there before.

Still not getting what the difference between the active and the focus states should be, if any.

Reply:
> Still not getting what the difference between the active and the focus states should be, if any.

I think @laremas should clarify on this.

Reply:
I've asked @laremas to chime in.

Reply:
The Active (button remains pressed) state has 1px solid cornflower blue border appearing and stays there when user is interacting with it. No transformation is needed now (meananing no animation will be rendered).

Reply:
It's not going to make a difference, as the hover state is already using the cornflower, so having the same border won't do much.

<img width="202" alt="Screenshot 2024-04-10 at 12 31 50" src="https://github.com/dusk-network/rusk/assets/34094428/471166e9-2f87-4682-afee-a94afaf08a9d">

The question I am trying to get an answer to is do we need an active state in the first place. I'm not sure it's needed...

Reply:
I already answered to you yesterday about it, I'll paste my answer here again though:

> I'll make an example
think of a settings button
in some cases you want the button to open a panel (drawer) while staying in the same page
in this case a switch isn't the correct fit
and here is where the button type="toggle" comes in
but it needs an active state that indicates, visually, that the button is in its pressed state
another common use case is a mute button in a video player
(in this case the active state is "special" because you often just change icon)
in a nutshell I think it's nice to have it ready

Reply:
Yes, but we don't have this use case. I removed the code related to the toggle.

And even if we did, why would this have a different appearance?

Reply:
> Yes, but we don't have this use case. I removed the code related to the toggle.

We don't have the use case *now*. The button is meant to be a generic component.
And, please, before removing wait for the others, especially @laremas , to barge in.

> And even if we did, why would an active/toggled button have a different appearance from the hover state?

Because they have a different meaning, so it's not always appropriate. But this, again, it's @laremas decision.

Reply:
> > Yes, but we don't have this use case. I removed the code related to the toggle.
> 
> We don't have the use case _now_. The button is meant to be a generic component. And, please, before removing wait for the others, especially @laremas , to barge in.
> 

My understanding is that this wasn't designed in the first place, so not sure why it was in the code to start off with.

Reply:
The all concern come from the fact that we wanted to avoid the use of a shading when the component is used, as it was a placeholder. We decided to do so cause it was the only place where a shading appeared in the design. the changes were just to change the shading to a 1px ( 0.0625em) cornflower blue solid border.

Reply:
I would also close this and open dedicated issues for the changes required, if anyit is hard to folllow the differents threads on this one. @Fulviuus

Reply:
> The all concern come from the fact that we wanted to avoid the use of a shading when the component is used, as it was a placeholder. We decided to do so cause it was the only place where a shading appeared in the design. the changes were just to change the shading to a 1px ( 0.0625em) cornflower blue solid border.

@nortonandreev wasn't referring to what was the focus state, but rather on the opportunity of having an active one.
I do think is needed (rationale above) even if there is no occurence of a "toggle button" in the wallet or explorer right now.
Also I don't see the need of removing the functionality which is already present and just needs a new style if we remove the transform.

Reply:
Ok i realized my comment was not related about the questions made. 

So i would put back the states as they was to cover all the possibilities. Also transformations as they were coded.

@nortonandreev We can def refine it later adding, changing or removing stuff. 

Generally i would avoid removing things here and there also if they appear to be useless at first glance, design grown besides being very minimal, all things live a balance where changing a little one will impact other as a tsunami.

Reply:
I am not willing to spend more time on this issue – progress with it as you wish.

I have done my best to refactor the code so that it's easy to follow and maintain, while using the design system in place as a source of truth on how the component should appear and behave.

> Also I don't see the need of removing the functionality which is already present

This is not a valid argument, as the functionality present was added as a matter of a personal preference and an imaginary use case, rather than what is actually needed.

> just needs a new style

If it was as simple as that, it wouldn't have taken so long to come up with a "new style". We are creating an issue that didn't exist in the first place.

---

I would also argue that a Button and Toggle are two different components with different purpose. To start off with, having the `active` prop within the button is confusing, as a button can be disabled, so the `active` attribute is ambiguous in this context. Second, code, such as the one below, which was present in the previous revision of the Button, screams an anti-pattern to me:

<img width="366" alt="Screenshot 2024-04-10 at 17 54 58" src="https://github.com/dusk-network/rusk/assets/34094428/c94da0ad-649d-4608-b54c-ebf43c184793">

If there is a code like this, perhaps something is wrong logically in the first place.



cc: @Fulviuus @kieranhall

Reply:
Let's do as @laremas suggests and put back the removed styles and move past this. As a follow up, @nortonandreev will propose a better solution in Duskit using multiple components that we can adopt after the design language foundations are established.

Reply:
> I would also close this and open dedicated issues for the changes required, if any. It is hard to folllow the different threads on this one. @Fulviuus

I've opened an [issue](https://github.com/dusk-network/duskit/issues/66) on Duskit.